libc/signal.c

```diff
@@ -0,0 +1,196 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ * HEADER:
+ *    - signal.h
+ *
+ * TESTED:
+ *    - psignal()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Mateusz Bloch
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <signal.h>
+#include <limits.h>
+#include <unistd.h>
+#include <unity_fixture.h>
+
+/* Typical error message does not exceed ~60 characters, that's why we expect a maximum value a little bit bigger */
+#define MAX_LEN_STRING 100
+#define PATH           "stderr.log"
+
+static FILE *err_file;
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-const-variable"
+static const int signal_codes[] = { SIGABRT, SIGALRM, SIGBUS, SIGCHLD, SIGCONT, SIGFPE, SIGHUP, SIGILL, SIGINT,
+	SIGKILL, SIGPIPE, SIGQUIT, SIGSEGV, SIGSTOP, SIGTERM, SIGTSTP, SIGTTIN, SIGTTOU, SIGUSR1, SIGUSR2, SIGPROF,
+	SIGSYS, SIGTRAP, SIGURG, SIGVTALRM, SIGXCPU, SIGXFSZ };
+
+static const unsigned int signal_codes_len = sizeof(signal_codes) / sizeof(signal_codes[0]);
+#pragma GCC diagnostic pop
+
+TEST_GROUP(signal_psignal);
+
+TEST_SETUP(signal_psignal)
+{
+	err_file = fopen(PATH, "a+");
+}
+
+
+TEST_TEAR_DOWN(signal_psignal)
+{
+	remove(PATH);
+}
+
+
+TEST(signal_psignal, basic)
+{
+/* Disabled because of #695 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/695 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#695 issue");
+#else
+	/* Since these function shall not return a value, we check only errno */
+	char buff[MAX_LEN_STRING];
+
+	for (int i = 0; i < signal_codes_len; i++) {
+		rewind(err_file);
+		memset(buff, 0, sizeof(buff));
+
+		TEST_ASSERT_NOT_NULL(freopen("stderr.log", "w", stderr));
```

Comment:
please check if it's possible to restore the `stderr` to the state before the test (right now all tests after this one would have `stderr` redirected to file) - probably to be done in `TEST_TEAR_DOWN`

Also: use define (currently `PATH`)